packages/cli/src/decorators/Scoped.ts

```diff
@@ -0,0 +1,35 @@
+import type { Scope } from '@n8n/permissions';
+import type { RouteScopeMetadata } from './types';
+import { CONTROLLER_ROUTE_SCOPES } from './constants';
+
+/**
+ * Decorator for a controller method to ensure the user has a scope,
+ * checking first at project level and then at global level.
+ *
+ * To check only at global level, use the `globalOnly` option.
+ *
+ * @example
+ * ```ts
+ * @RestController()
+ * export class WorkflowController {
+ *   @Get('/:id')
+ *   @Scoped('workflow:read')
+ *   async getWorkflow(req, res) { ... }
+ * }
+ * ```
+ */
+export const Scoped = (scope: Scope | Scope[], { globalOnly } = { globalOnly: false }) => {
```

Comment:
Can we please extract the type for the options into an interface, and add a JSDoc for `globalOnly`? that way anyone reading this code can better understand what this option means.

or perhaps break this up into two separate decorators, instead of conflating two different things.
We could possible merge `@Authorized` and `@RequireGlobalScope` into a `@GlobalScope`, and create a separate decorator `@LocalScope` or `@ProjectScope` for the granular access control.